--- a/disposable/index.mjs
+++ b/disposable/index.mjs
@@ -1,6 +1,6 @@
 /// <reference path="./core-js-pure.d.ts" />
-import CoreJsDisposableStack from 'core-js-pure/actual/disposable-stack';
-import CoreJsAsyncDisposableStack from 'core-js-pure/actual/async-disposable-stack';
+import CoreJsDisposableStack from 'core-js-pure/actual/disposable-stack/index.js';
+import CoreJsAsyncDisposableStack from 'core-js-pure/actual/async-disposable-stack/index.js';
 
 /**
  * Well-known `Symbol.dispose`. Resolves to the native well-known symbol when
