diff --git a/dump.go b/dump.go
index c6c24ed..c21e2c2 100644
--- a/dump.go
+++ b/dump.go
@@ -7,6 +7,7 @@ import (
 	"io"
 	"net"
 	"net/http"
+	"time"
 )

 var _ WrapResponseWriter = (*bodyWriter)(nil)
@@ -60,6 +61,15 @@ func (w *bodyWriter) Flush() {
 	}
 }

+func (w bodyWriter) SetWriteDeadline(t time.Time) error {
+	if d, ok := w.ResponseWriter.(interface {
+		SetWriteDeadline(t time.Time) error
+	}); ok {
+		return d.SetWriteDeadline(t)
+	}
+	return nil
+}
+
 // implements http.Hijacker
 func (w *bodyWriter) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	if hi, ok := w.ResponseWriter.(http.Hijacker); ok {
