diff --git a/dump.go b/dump.go
index b5cb717..1d73f03 100644
--- a/dump.go
+++ b/dump.go
@@ -38,7 +38,7 @@ func (w *bodyWriter) Write(b []byte) (int, error) {
 	if w.body != nil {
 		if w.body.Len()+len(b) > w.maxSize {
 			w.body.Truncate(min(w.maxSize, len(b), w.body.Len()))
-			w.body.Write(b[:w.maxSize-w.body.Len()])
+			w.body.Write(b[:min(w.maxSize-w.body.Len(), len(b))])
 		} else {
 			w.body.Write(b)
 		}
